### PR TITLE
Refactor: src/screens/OrganizationDashboard from Jest to Vitest

### LINKED ISSUE #2558: Refactor: `src/screens/OrganizationDashboard` from Jest to Vitest
**Is your feature request related to a problem? Please describe.**
This file is currently tested using Jest. As part of our migration to Vitest, we need to refactor this file's test cases to align with Vitest's syntax and features.

**Acceptance Criteria**
1. Replace Jest-specific functions and mocks with Vitest equivalents
2. Ensure all tests in `src/screens/OrganizationDashboard` pass after migration using `npm run test:vitest`
3. Maintain the test coverage for the file as 100% after migration
4. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional context**

- Refer to the merged PR for the base Vitest configuration:
https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
- Vitest documentation:
https://vitest.dev/

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by abbi4code**
**What kind of change does this PR introduce?**
This PR migrates the OrganizationDashboard tests from Jest to Vitest and addresses all failing tests to ensure they pass successfully

**Issue Number:**

Fixes #2558 

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**
Screenshot from 2024-12-18 14-54-12

**Reason for Adding identity-obj-proxy for CSS Modules:**

The purpose of adding the identity-obj-proxy mapping and the CSS transformation is to properly parse CSS Modules in Jest.However, CSS Modules by default may still get ignored by Jest's configuration. To ensure proper handling, we need to map CSS Modules to identity-obj-proxy, which allows Jest to mock and return class names as string values. This avoids errors in Jest tests and ensures that styles are correctly interpreted in the test environment.

- ref- https://www.jest-preview.com/docs/getting-started/installation/

**Alternative Approach with jest-css-modules-transform**
I also got a alternative approach in which i can replace the below configuration with jest-css-modules-transform, which specifically targets CSS Modules. 

```
  transform: {
    '^.+\\.(css|scss|sass|less)$': 'jest-preview/transforms/css', // CSS transformations'
  },
  moduleNameMapper: {
    '\\.(css|less)$': 'identity-obj-proxy',
  },
```
with 
```
transform: {
  '^.+\\.(css|scss|sass|less)$': 'jest-css-modules-transform',  
  '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': 'jest-preview/transforms/file', 
}
```

Replacing ```jest-preview/transforms/css``` with ```jest-css-modules-transform```:
The key difference with this approach is that ``` jest-css-modules-transform ``` is specifically designed for transforming , parsing and mocking CSS Modules in a Jest environment. It automatically handles CSS Modules by transforming them into mockable objects, which eliminates the need for using ```identity-obj-proxy```.

**Test Cases Results(Using 1st approach)**

Screenshot from 2024-12-18 14-50-24

## Summary by CodeRabbit

- **Tests**
	- Enhanced the test suite for the `OrganizationScreen` component with improved assertions using module-scoped styles.
	- Updated the `OrganizationDashboard` component tests to use Vitest for mocking, with detailed comments and structured test cases for various scenarios.

### CODE REVIEW

**Comment by abbi4code** [comment]
@varshith257 I have fixed all the failing tests . Could you please review it ?

File: jest.config.js
[the remark this answers is not in the document]
Comment (abbi4code) [reply jest.config.js:1]: I have removed the changes but i noticed that the changes in PR #2662 are identical to my commit. I’m still glad that all the failed test cases are fixed now, but it seems that my work was copied without proper attribution.

[the remark this answers is not in the document]
Comment (abbi4code) [reply jest.config.js:1]: > Can you remove changes in this file?

I have removed the changes but i noticed that the changes in PR #2662 are identical to my commit. I’m still glad that all the failed test cases are fixed now, but it seems that my work was copied without proper attribution.

**Comment by abbi4code** [comment]
@coderabbitai review

**Review by palisadoes — APPROVED** [review]

**Pull request merged**